xplique/commons/torch_operations.py

```diff
@@ -0,0 +1,150 @@
+"""
+Custom pytorch operations
+"""
+
+from abc import ABC, abstractmethod
+from math import ceil
+import numpy as np
+import torch
+
+from ..types import Union, Tuple, Callable, List, Dict
+
+
+class NlpPreprocessor(ABC):
+    """
+    Abstract base class for NLP preprocessing.
+
+    Parameters
+    ----------
+    tokenizer
+        The tokenizer function to be used for tokenization. It must be
```

Comment:
This description needs to be clarified as no model is defined here. Does this correspond to what you define in the nlp_commons?

If it corresponds to the tokenizer-model pair, as in HuggingFace, then I recommend having a class that treats them both at the same time. It is easier to understand and the API would be more natural.